tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+	}
+
+	private function _addOneDocumentToIndex() {
+		$document = new Opus_Document( $this->document_id );
+		$this->indexer->addDocumentsToIndex( $document );
+	}
+
+	/**
+	 *
+	 * @param string $filename
+	 * @param string $label
+	 */
+	private function _addFileToDocument( $filename, $label ) {
+		$doc  = new Opus_Document( $this->document_id );
+		$file = $doc->addFile();
+		$file->setTempFile( APPLICATION_PATH . '/tests/fulltexts/' . $filename );
+		$file->setPathName( $filename );
+		$file->setLabel( $label );
+		$file->setVisibleInFrontdoor( '1' );
+
+		$doc->store();
+
+		$this->indexer->addDocumentsToIndex( $doc );
+	}
+
+	public function testAttachFulltextToNull() {
+		$this->markTestSkipped( 'not supported in Opus_Search API' );
+
+		// TODO is this intended behaviour of a unit test?
+		// apply a hack to be able to test a private method directly
+		$class  = new ReflectionClass( 'Opus_SolrSearch_Index_Indexer' );
+		$method = $class->getMethod( 'attachFulltextToXml' );
+		$method->setAccessible( true );
+
+		$indexer = new Opus_SolrSearch_Index_Indexer();
+		$method->invokeArgs( $indexer, array( new DomDocument(), null, 1 ) );
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 */
+	public function testIndexDocumentWithMultipleTitleMainInSameLanguage() {
+		$doc = new Opus_Document();
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+
+		$title = new Opus_Title();
+		$title->setValue( 'foo' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleMain( $title );
+
+		$title = new Opus_Title();
+		$title->setValue( 'bar' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleMain( $title );
+
+		$doc->store();
+
+		$exception = null;
+		try {
+			$this->indexer->addDocumentsToIndex( $doc );
+		}
+		catch ( Exception $e ) {
+			$exception = $e;
+		}
+
+		$this->assertNotNull( $exception );
+		$this->assertInstanceOf( 'Opus_Search_InvalidQueryException', $exception );
+
+		$doc->deletePermanent();
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 */
+	public function testIndexDocumentWithMultipleAbstractsInSameLanguage() {
+		$doc = new Opus_Document();
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+
+		$title = new Opus_Title();
+		$title->setValue( 'foo' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleAbstract( $title );
+
+		$title = new Opus_Title();
+		$title->setValue( 'bar' );
+		$title->setLanguage( 'eng' );
+		$doc->addTitleAbstract( $title );
+
+		$doc->store();
+
+		$exception = null;
+		try {
+			$this->indexer->addDocumentsToIndex( $doc );
+		}
+		catch ( Exception $e ) {
+			$exception = $e;
+		}
+
+		$this->assertNotNull( $exception );
+		$this->assertInstanceOf( 'Opus_Search_InvalidQueryException', $exception );
+
+		$doc->deletePermanent();
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2240
+	 *
+	 * @expectedException Opus_SolrSearch_Index_Exception
+	 */
+	public function testIndexDocumentWithUnknownIndexField() {
```

Comment:
What is the new strategy here? What could it be? The ticket OPUSVIER-2240 was basically about mismatches between XML send to Solr and the schema. Single value fields with multiple values of fields in the XML that weren't supported in the schema. How can we catch that? Right now Solr Logging is showing such problems with our unit tests. Is there anything to catch?